domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -3158,4 +3799,72 @@ public static String getMockData(String path) throws IOException {
       }
     }
   }
+
+  @Test
+  public void testCacheResultCQLWrapper(TestContext context) {
+    createNumbers(context, 1, 2, 3);
+
+    postgresClient.removePersistentCacheResult("cache_numbers_does_not_exist",
+        context.asyncAssertFailure());
+
+    postgresClient.persistentlyCacheResult("cache_numbers",
+        "SELECT i FROM numbers WHERE i IN (1, 3, 5) ORDER BY i", context.asyncAssertSuccess(
+            res -> {
+              context.assertEquals(2, res);
+              postgresClient.removePersistentCacheResult("cache_numbers",
+                  context.asyncAssertSuccess());
+            }
+        ));
+  }
+
+  @Test
+  public void selectReturnFail(TestContext context) {
+    Promise<RowSet<Row>> promise = Promise.promise();
+    promise.complete(null);
+    PostgresClient.selectReturn(promise.future(), context.asyncAssertFailure());
+  }
+
+  @Test
+  public void selectReturnEmptySet(TestContext context) {
+    RowSet rowSet = new FakeRowSet(0);
+    Promise<RowSet<Row>> promise = Promise.promise();
+    promise.complete(rowSet);
+    PostgresClient.selectReturn(promise.future(), context.asyncAssertSuccess(res ->
+      context.assertEquals(null, res)));
+  }
+
+  @Test
+  public void selectReturnOneRow(TestContext context) {
+    List<String> columns = new LinkedList<>();
+    columns.add("field");
+    RowDesc rowDesc = new RowDesc(columns);
+    List<Row> rows = new LinkedList<>();
+    Row row = new RowImpl(rowDesc);
+    row.addString("value");
+    rows.add(row);
+    RowSet rowSet = new FakeRowSet(1).withColumns(columns).withRows(rows);
+
+    Promise<RowSet<Row>> promise = Promise.promise();
+    promise.complete(rowSet);
+    PostgresClient.selectReturn(promise.future(), context.asyncAssertSuccess(res ->
+        context.assertEquals("value", res.getString(0))));
+  }
+
+  @Test
+  public void getTotalRecordsTest() {
```

Comment:
This should go into PostgresClientTest because it is a static method that does not require a database connection.

Reply:
ok